### PR TITLE
RC-34353 Break Glass Access

### PR DESCRIPTION
**Pull request by ankitrafay**

### CODE REVIEW

File: examples/resources/rafay_break_glass_access/variables.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+variable "rafay_config_file" {
+  description = "rafay provider config file for authentication"
+  sensitive   = true
+  default     = "/Users/ankit/Downloads/Rafay_System-ankit@rafay.co.json"
```
Comment (saisrikar-rafay) [inline examples/resources/rafay_break_glass_access/variables.tf:4]: Please avoid using personal details

File: examples/resources/rafay_break_glass_access/resource.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
+resource "rafay_breakglassaccess" "test_user" {
+  metadata {
+    name = "ankit+terraform@rafay.co"
```
Comment (saisrikar-rafay) [inline examples/resources/rafay_break_glass_access/resource.tf:3]: Please avoid using personal details

File: examples/resources/rafay_groupassociation/data.tf [hunk 3]
```diff
@@ -1,6 +1,6 @@
 data "rafay_groupassociation" "association" {
-    group    = "group-name"
-    project = "demo"
+    group    = "rctl-grp"
+    project = "sample-shetty"
```
Comment (saisrikar-rafay) [inline examples/resources/rafay_groupassociation/data.tf:3]: Please avoid using personal details

File: examples/resources/rafay_groupassociation/providers.tf [hunk 4]
```diff
@@ -2,14 +2,14 @@ terraform {
   required_providers {
     rafay = {
       version = ">= 0.1"
-      source  = "registry.terraform.io/RafaySystems/rafay"
+      source  = "registry.terraform.io/Rafay/rafay"
```
Comment (saisrikar-rafay) [inline examples/resources/rafay_groupassociation/providers.tf:5]: Looks to be a temporary change

File: examples/resources/rafay_groupassociation/providers.tf [hunk 5]
```diff
@@ -2,14 +2,14 @@ terraform {
...
     }
   }
 }
 
 variable "rafay_config_file" {
   description = "rafay provider config file for authentication"
-  default     = "<rafay-config-json-file>"
+  default     = "/Users/ankit/Downloads/tf.json"
```
Comment (saisrikar-rafay) [inline examples/resources/rafay_groupassociation/providers.tf:12]: Looks to be a temporary change

File: examples/resources/rafay_groupassociation/resource.tf [hunk 6]
```diff
@@ -1,22 +1,49 @@
...
+  name        = "terraform-group4"
+  description = "dev1-description"
+}
+
+
 resource "rafay_groupassociation" "groupassociation" {
-  group      = "dev1"
-  project    = "terraform"
+  group      = rafay_group.group1.name
+  project      = "a-ankit1"
```
Comment (saisrikar-rafay) [inline examples/resources/rafay_groupassociation/resource.tf:21]: Please avoid using personal details

File: go.mod [hunk 7]
```diff
@@ -218,10 +218,10 @@ require (
 
 replace (
 	cloud.google.com/go => cloud.google.com/go v0.100.2
-	// github.com/RafaySystems/rafay-common => ../rafay-common
+	 github.com/RafaySystems/rafay-common => ../rafay-common
```
Comment (saisrikar-rafay) [inline go.mod:221]: TODO - please replace the dependencies

**Review by saisrikar-rafay — CHANGES_REQUESTED** [review]

File: examples/resources/rafay_break_glass_access/resource.tf (hunk 2 above)
Comment (ankitrafay) [reply examples/resources/rafay_break_glass_access/resource.tf:3]: modified this

File: examples/resources/rafay_groupassociation/data.tf (hunk 3 above)
Comment (ankitrafay) [reply examples/resources/rafay_groupassociation/data.tf:3]: reverted this change

File: examples/resources/rafay_break_glass_access/variables.tf (hunk 1 above)
Comment (ankitrafay) [reply examples/resources/rafay_break_glass_access/variables.tf:4]: modified this

File: examples/resources/rafay_groupassociation/providers.tf (hunk 5 above)
Comment (ankitrafay) [reply examples/resources/rafay_groupassociation/providers.tf:12]: reverted this change

File: examples/resources/rafay_groupassociation/resource.tf (hunk 6 above)
Comment (ankitrafay) [reply examples/resources/rafay_groupassociation/resource.tf:21]: reverted this change

File: go.mod (hunk 7 above)
Comment (ankitrafay) [reply go.mod:221]: will replace once common is merged

File: examples/resources/rafay_break_glass_access/providers.tf [hunk 8]
```diff
@@ -0,0 +1,12 @@
+terraform {
+  required_providers {
+    rafay = {
+      version = ">= 0.1"
+      source  = "registry.terraform.io/Rafay/rafay"
```
Comment (krishnarafay) [inline examples/resources/rafay_break_glass_access/providers.tf:5]: it should be `registry.terraform.io/RafaySystems/rafay`

Comment (ankitrafay) [reply examples/resources/rafay_break_glass_access/providers.tf]: updated it

File: examples/resources/rafay_break_glass_access/resource.tf [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  spec {
+    groups {
+      group_expiry {
+        expiry     = 7
+        name      = "grp3"
+      }
+      group_expiry {
+        expiry     = 8
+        name      = "grp1"
+        start_time = "2024-09-20T08:00:00Z"
```
Comment (krishnarafay) [inline examples/resources/rafay_break_glass_access/resource.tf:14]: is it need at post API call..?

**Review by krishnarafay — APPROVED** [review]

**Pull request merged**